src/main/java/com/fasterxml/jackson/databind/deser/SettableAnyProperty.java

```diff
@@ -141,8 +145,31 @@ public Object deserialize(JsonParser p, DeserializationContext ctxt) throws IOEx
     public void set(Object instance, String propName, Object value) throws IOException
     {
         try {
-            // note: can not use 'setValue()' due to taking 2 args
-            _setter.getAnnotated().invoke(instance, propName, value);
+			// note: can not use 'setValue()' due to taking 2 args
+			if (_setter instanceof AnnotatedMethod) {
+				((AnnotatedMethod) _setter).getAnnotated().invoke(instance, propName, value);
+			}
+			// if annotation in the field (only map is supported now)
+			else if (_setter instanceof AnnotatedField) {
+				AnnotatedField field = ((AnnotatedField) _setter);
+				// check whether the annotated field is an instance of Map, if
+				// not throw Error
+				if (!Map.class.isAssignableFrom(field.getRawType())) {
```

Comment:
I think this should fail earlier, on construction (or perhaps introspection), no point in waiting until attempt is made to set it.